enterprise/frontend/src/metabase-enterprise/whitelabel/components/ChartColorSettings/utils.ts

```diff
@@ -46,12 +46,18 @@ const getAutoColors = (
   oldColors: (Color | undefined)[],
   fallbackColor: Color,
 ) => {
-  const baseColor = oldColors.find(color => color != null) ?? fallbackColor;
+  const oldColor = oldColors.find(color => color != null);
 
   const autoColors: Color[] = [];
-  oldColors.forEach((_, index) =>
-    autoColors.push(getNextColor(index ? autoColors[index - 1] : baseColor)),
-  );
+  oldColors.forEach((_, index) => {
+    if (index > 0) {
+      autoColors.push(getNextColor(autoColors[index - 1]));
+    } else if (oldColor) {
+      autoColors.push(getNextColor(oldColor));
+    } else {
+      autoColors.push(fallbackColor);
+    }
```

Comment:
nit: This reads a bit hard. Normally, I could imagine we do something with the array starting at index 0, then do another thing, and so on.

But this shows the condition from index 1 and more first. So, if I were to follow how this works in the first loop the code in the 2nd or 3rd loop would be executed. I'd expect the last 2 conditions to be first as it would feel more natural.

Reply:
Refactored, thanks!